src/classes/upsert-session.ts

```diff
@@ -0,0 +1,24 @@
+import { Session } from './session';
+import { TUpsertOptions } from '../types/upsert-options';
+import { ISequelizeService } from '../interfaces/sequelize-service';
+import { IUpsertSession } from '../interfaces/upsert-session';
+
+export class UpsertSession<W extends {}, R extends W, C extends {}, KC extends keyof C, O extends {} = {}> extends Session<W, R, C, TUpsertOptions<R, C, KC> & O> implements IUpsertSession<W, R, C, KC, O> {
+  private created: boolean | undefined = false;
```

Comment:
I only know after the upsert if the row was created or updated, hence the `| undefined`.